runtime/compiler/optimizer/OSRGuardInsertion.cpp

```diff
@@ -386,9 +386,9 @@ void TR_OSRGuardInsertion::removeHCRGuards(TR_BitVector &fearGeneratingNodes, TR
 
             // if virtual guards kill fear we can short cut additional analysis if the method still has a guard
             // we can mark that guard as an OSR guard and continue without needing a data flow analysis
-            if (TR_FearPointAnalysis::virtualGuardsKillFear(comp())
-                && additionalVirtualGuard
-                && comp()->cg()->supportsMergingGuards())
+            if (additionalVirtualGuard
+                  && TR_FearPointAnalysis::virtualGuardsKillFear(comp(), potentialGuard)
+                  && comp()->cg()->supportsMergingGuards())
```

Comment:
Could you please move this `supportsMergingGuards()` condition into `virtualGuardsKillFear()`? e.g.
```
if (!comp->cg()->supportsMergingGuards())
   return false;
```
Then we wouldn't need to `&&` at (almost) every call site

Reply:
Added the check in the beginning of virtualGuardKillFear function and removed the check from all callers of that function in https://github.com/eclipse-openj9/openj9/pull/16354/commits/65dee583088499cc7e1787e743d0191096fae72e